sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala

```diff
@@ -110,6 +108,12 @@ class HiveContext private[hive](
       isRootContext = false)
   }
 
+  private val hiveSessionState = new HiveSessionState(self)
+
+  protected[sql] override val sessionState = hiveSessionState
+
+  protected[hive] val hcatalog = hiveSessionState.metastoreCatalog
```

Comment:
hcatalog is a bad name because there is actually a hive-related project called hcatalog. Maybe just call this hiveCatalog?